src/main/java/org/opensearch/security/support/Base64CustomHelper.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.support;
+
+import com.amazon.dlic.auth.ldap.LdapUser;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.io.BaseEncoding;
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.Nullable;
+import org.opensearch.core.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.common.Strings;
+import org.opensearch.security.auth.UserInjector;
+import org.opensearch.security.user.User;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+import static org.opensearch.security.support.SafeSerializationUtils.prohibitUnsafeClasses;
+
+/**
+ * Provides support for Serialization/Deserialization of objects of supported classes into/from Base64 encoded stream
+ * using the OpenSearch's custom serialization protocol implemented by the StreamInput/StreamOutput classes.
+ */
+public class Base64CustomHelper {
+
+    private enum CustomSerializationFormat {
+
+        WRITEABLE(1),
+        STREAMABLE(2),
+        GENERIC(3);
+
+        private final int id;
+
+        CustomSerializationFormat(int id) {
+            this.id = id;
+        }
+
+        static CustomSerializationFormat fromId(int id) {
```

Comment:
Nit: Not sure I follow the naming format of this method... I assume it would be used as `Base64CustomHelper.fromId(<>)` but this name is not intuitive in my opinion. You are making a `CustomSerializationFormat` but no part of the method references the outcome.

Reply:
It'll be used as `CustomSerializationFormat.fromId(<>)`. It returns the `CustomSerializationFormat` enum corresponding to the integer id.